src/lifecycles/load.js

```diff
@@ -22,6 +22,10 @@ import { assign } from "../utils/assign.js";
 
 export function toLoadPromise(app) {
   return Promise.resolve().then(() => {
+    if (app.loadPromise) {
```

Comment:
We store the loadPromise on the app object itself so that multiple runs of reroute are able to share the same loadPromise.